src/Text/Pandoc/Writers/JATS/References.hs

```diff
@@ -0,0 +1,125 @@
+                => WriterOptions
+                -> Reference Inlines
+                -> JATS m (Doc Text)
+referenceToJATS _opts ref = do
+  let refType = referenceType ref
+  let pubType = [("publication-type", refType) | not (T.null refType)]
+  let wrap = inTags True "ref" [("id", "ref-" <> unItemId (referenceId ref))]
+           . inTags True "element-citation" pubType
+  return . wrap . vcat $
+    [ authors
+    , "title" `varInTag`
+      if refType == "book"
+      then "source"
+      else "article-title"
+    , if refType == "book"
+      then empty
+      else "container-title" `varInTag` "source"
+    , editors
+    , "publisher"       `varInTag` "publisher-name"
+    , "publisher-place" `varInTag` "publisher-loc"
+    , yearTag
+    , accessed
+    , "volume"          `varInTag` "volume"
+    , "issue"           `varInTag` "issue"
+    , "page-first"      `varInTag` "fpage"
+    , "page-last"       `varInTag` "lpage"
+    , "pages"           `varInTag` "page-range"
+    , "ISBN"            `varInTag` "isbn"
+    , "ISSN"            `varInTag` "issn"
+    , varInTagWith "doi"  "pub-id" [("pub-id-type", "doi")]
+    , varInTagWith "pmid" "pub-id" [("pub-id-type", "pmid")]
+    ]
+  where
+    varInTag var tagName =
+      case lookupVariable var ref >>= valToText of
+        Nothing  -> mempty
+        Just val -> inTags' tagName [] $ literal val
+
+    varInTagWith var tagName tagAttribs =
+      case lookupVariable var ref >>= valToText of
+        Nothing  -> mempty
+        Just val -> inTags' tagName tagAttribs $ literal val
+
+    authors = inTags' "person-group" [("person-group-type", "author")] $
+      case lookupVariable "author" ref of
+        Just (NamesVal names) -> vcat $ map toNameElement names
+        _                     -> empty
+
+    editors = inTags' "person-group" [("person-group-type", "editor")] $
+      case lookupVariable "editor" ref of
+        Just (NamesVal names) -> vcat $ map toNameElement names
+        _                     -> empty
+
+    yearTag =
+      case lookupVariable "issued" ref of
+        Just (DateVal date) -> toYearElement date
+        _ -> empty
+
+    accessed =
+      case lookupVariable "accessed" ref of
+        Just (DateVal d) -> inTags' "date-in-citation"
+                                    [("content-type", "access-date")]
+                                    (toYearElement d)
+        _ -> empty
+
+toYearElement :: Date -> Doc Text
+toYearElement date =
+  case dateParts date of
+    (DateParts (year:_):_) ->
+      let yeartxt = tshow year
+      in inTags' "year" [("iso-8601-date", yeartxt)] $ literal yeartxt
+    _                      -> empty
+
+toNameElement :: Name -> Doc Text
+toNameElement name = inTags' "name" [] . mconcat $
+  [ maybe mempty (inTags' "surname"     [] . literal) (nameFamily name)
+  , maybe mempty (inTags' "given-names" [] . literal) (nameGiven name)
+  , maybe mempty (inTags' "suffix"      [] . literal) (nameSuffix name)
+  ]
```

Comment:
In addition to "suffix" many names have "dropping-particle" and "non-dropping-particle."
If JATS doesn't have elements for this, I think you  want to add dropping-particle to the end of given-names and add non-dropping-particle to the beginning of surname (with spaces separating).

Reply:
JATS only has a [`<prefix>` element](https://jats.nlm.nih.gov/publishing/tag-library/1.2/element/prefix.html), but it's intended usage are titles and honorifics.

After some reading, I believe the correct method to use [`<name-alternatives>`](https://jats.nlm.nih.gov/publishing/tag-library/1.3d1/element/name-alternatives.html) with different versions for display and indexing. Then we can also add a `<string-name>` containing the nameLiteral field.

Cc @hubgit, who probably could offer deeper insights into proper, JATS4R conformant, tagging.

Reply:
Note that nameLiteral is generally only used when the other fields are not, so that's not going to work.

I think the best thing to do is what I suggested:  when particles are not explicitly set off in their own fields, they are generally incorporated into given names (dropping) and surnames (nondropping).

Reply:
Oh, and you'll also have to handle the case where nameLiteral has a value but the other fields don't.

Reply:
Ok, done.

I just noted that all my tests are local, will add them tomorrow.

Reply:
There's not too much about names in [the JATS4R recommendations for citations](https://jats4r.org/citations):

> `<name>` / `<string-name>` / `<collab>`. The `<name> `or` <string-name>` element should be used for an individual named author. Use `<name>` when the ancestor element is `<element-citation>`, and `<string-name>` when the parent is `<mixed-citation>`.

> When groups of authors (consortia) are cited under a group name or an institution is cited as the contributor, the `<collab>` element should be used.

The [PMC tagging guidelines](https://www.ncbi.nlm.nih.gov/pmc/pmcdoc/tagging-guidelines/article/dobs.html#dob-refs) have some examples of citations where `<given-names>` or `<surname>` includes text that might be in the "dropping-particle"/"non-dropping-particle" data:

```
6. (a) De Albuquerque IL, Galeffi C, Casinovi CG, Marini-Bettolo GB. Gazz Chim Ital. 1964;94:287. (b) Alcantara, A. F. de C.; Souza, MR.; Pilo-Veloso, D. Fitoterapia. 2000;71:613.
```

```xml
<ref id="R6">
  <label>6</label>
  <element-citation publication-type="journal">
    <label>(a)</label>
    <name>
      <surname>De Albuquerque</surname>
      <given-names>I. L.</given-names>
    </name>
    <name>
      <surname>Galeffi</surname>
      <given-names>C.</given-names>
    </name>
    <name>
      <surname>Casinovi</surname>
      <given-names>C. G.</given-names>
    </name>
    <name>
      <surname>Marini-Bettolo</surname>
      <given-names>G. B.</given-names>
    </name>
    <source>Gazz. Chim. Ital.</source>
    <year>1964</year>
    <volume>94</volume>
    <fpage>287</fpage>
  </element-citation>
  <element-citation publication-type="journal">
    <label>(b)</label>
    <name>
      <surname>Alcantara</surname>
      <given-names>A. F. de C.</given-names>
    </name>
    <name>
      <surname>Souza</surname>
      <given-names>M. R.</given-names>
    </name>
    <name>
      <surname>Pilo-Veloso</surname>
      <given-names>D.</given-names>
    </name>
    <source>Fitoterapia</source>
    <year>2000</year>
    <volume>71</volume>
    <fpage>613</fpage>
  </element-citation>
</ref>
```